moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -274,8 +274,38 @@ void robot_trajectory::RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::Robot
     {
       trajectory.multi_dof_joint_trajectory.points[i].transforms.resize(mdof.size());
       for (std::size_t j = 0; j < mdof.size(); ++j)
+      {
         tf::transformEigenToMsg(waypoints_[i]->getJointTransform(mdof[j]),
                                 trajectory.multi_dof_joint_trajectory.points[i].transforms[j]);
+        if (waypoints_[i]->hasVelocities())
+        {
+          const std::vector<std::string> names = mdof[j]->getVariableNames();
```

Comment:
I don't currently have anything to test a floating joint but it will likely come into my scope soon enough. I have added the todo and will get back to it another time.